contracts/token/ERC1155/extensions/ERC1155URIStorage.sol

```diff
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: MIT
+// OpenZeppelin Contracts v4.4.1 (token/ERC1155/extensions/ERC1155URIStorage.sol)
+
+pragma solidity ^0.8.0;
+
+import "../../../utils/Strings.sol";
+import "../ERC1155.sol";
+import "./ERC1155Supply.sol";
+
+/**
+ * @dev ERC1155 token with storage based token URI management.
+ * Inspired by the ERC721URIStorage extension
+ */
+abstract contract ERC1155URIStorage is ERC1155Supply {
+    using Strings for uint256;
+
+    // Optional mapping for token URIs
+    mapping(uint256 => string) private _tokenURIs;
+
+    /**
+     * @dev {IERC721Metadata-tokenURI} token URIs for ERC1155 tokens.
+     */
+    function tokenURI(uint256 tokenId) public view virtual returns (string memory) {
+        require(exists(tokenId), "ERC1155URIStorage: URI query for nonexistent token");
+
+        string memory _tokenURI = _tokenURIs[tokenId];
+        string memory base = uri(0);
+
+        // If there is no base URI, return the token URI.
+        if (bytes(base).length == 0) {
+            return _tokenURI;
+        }
+        // If both are set, concatenate the base URI and tokenURI (via abi.encodePacked).
+        if (bytes(_tokenURI).length > 0) {
+            return string(abi.encodePacked(base, _tokenURI));
+        }
+
+        // If there is a base URI set but no tokenURI, return the base URI.
+        return base;
+    }
+
+    /**
+     * @dev Sets `_tokenURI` as the tokenURI of `tokenId`.
+     *
+     * Requirements:
+     *
+     * - `tokenId` must exist.
+     */
+    function _setTokenURI(uint256 tokenId, string memory _tokenURI) internal virtual {
+        require(exists(tokenId), "ERC1155URIStorage: URI set of nonexistent token");
+        _tokenURIs[tokenId] = _tokenURI;
```

Comment:
You need to emit `event URI(string _value, uint256 indexed _id);` as described in the [ERC](https://eips.ethereum.org/EIPS/eip-1155)

Reply:
Good catch! Thanks for that, I added a new [commit](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/3210/commits/0cbffd5be0ef7aa88b4380d80e081b0c6545f2e3).